Lib/ntpath.py

```diff
@@ -146,29 +146,55 @@ def splitdrive(p):
             sep = b'\\'
             altsep = b'/'
             colon = b':'
+            prefix = b'\\\\?\\'
+            unc_prefix = b'UNC\\'
+            vol_prefix = b'VOLUME'
         else:
             sep = '\\'
             altsep = '/'
             colon = ':'
+            prefix = '\\\\?\\'
+            unc_prefix = 'UNC\\'
+            vol_prefix = 'VOLUME'
         normp = p.replace(altsep, sep)
-        if (normp[0:2] == sep*2) and (normp[2:3] != sep):
-            # is a UNC path:
-            # vvvvvvvvvvvvvvvvvvvv drive letter or UNC path
-            # \\machine\mountpoint\directory\etc\...
-            #           directory ^^^^^^^^^^^^^^^
-            index = normp.find(sep, 2)
-            if index == -1:
+        if normp[:4] == prefix:
+            if normp[4:8].upper() == unc_prefix:
+                # e.g. \\?\UNC\server\share\dir\file
+                start = 8
+            elif normp[4:10].upper() == vol_prefix:
```

Comment:
```suggestion
            elif normp[4:11].upper() == vol_prefix:
```
More generically, use `normp[4:4 + len(vol_prefix)]`.